ossfuzz/decompress_fuzzer.c

```diff
@@ -9,13 +9,13 @@
 #include <string.h>
 
 #include "fuzz_helpers.h"
+#include "fuzz_data_producer.h"
 #include "lz4.h"
 
 int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 {
-
-    uint32_t seed = FUZZ_seed(&data, &size);
-    size_t const dstCapacity = FUZZ_rand32(&seed, 0, 4 * size);
+    FUZZ_dataProducer_t *producer = FUZZ_dataProducer_create(data, size);
+    size_t const dstCapacity = FUZZ_dataProducer_uint32(producer, 0, 4 * size);
```

Comment:
After this we'll want:

```c
    FUZZ_buffer_t const buffer = FUZZ_dataProducer_remaining(producer);
    size_t const smallDictSize = buffer.size + 1;
    // use buffer instead of (data, size)...
```

Right now, we are using the last few bytes twice. Once to get `dstCapacity`, and once as the input to the decompression functions. We should only use each byte in one way. We can do that with a `FUZZ_dataProducer_remaining()` function that returns the unused bytes.

We can use the remaining bytes in each of the `LZ4_decompress*()` functions because they interpret the bytes in the same way, since they all accept the same format.

Reply:
I think only the remaining bytes are being fed into the decompression functions because I change size here: https://github.com/lz4/lz4/pull/779/files/fad8c97532f74d92f6aa4427a739610035fcbbd1..a9ac05645644a0615b558f6ac655c4ae46c4a926#diff-7d201b931de5a277000ee0a58cff72abR29

Or maybe you're just saying it would be clearer to do that in a function FUZZ_dataProducer_remaining()?

Reply:
Good point, I missed that.